test/test_ion_stream.cpp

```diff
@@ -164,23 +164,23 @@ TEST(IonStream, WriteToUserStream) {
 
     ION_ASSERT_OK(ion_writer_start_container(writer, tid_STRUCT));
     ION_STRING fieldNameString;
-    ion_string_assign_cstr(&fieldNameString, "str_col1", strlen("str_col1"));
+    ion_string_assign_cstr(&fieldNameString, (char *)"str_col1", strlen("str_col1"));
     ION_ASSERT_OK(ion_writer_write_field_name(writer, &fieldNameString));
 
     ION_STRING value_string;
-    ion_string_assign_cstr(&value_string, "str_val1", strlen("str_val1"));
+    ion_string_assign_cstr(&value_string, (char *)"str_val1", strlen("str_val1"));
     ION_ASSERT_OK(ion_writer_write_string(writer, &value_string));
 
-    ion_string_assign_cstr(&fieldNameString, "str_col2", strlen("str_col2"));
+    ion_string_assign_cstr(&fieldNameString, (char *)"str_col2", strlen("str_col2"));
```

Comment:
Is there a reason not to change `ion_string_assign_cstr` take a `const char *`? Wouldn't that be preferable?

Reply:
That would be preferred, but I didn't want to make a potentially API breaking change. I need to dig deeper into what, if any, impact changing a const would have on a C API.

Reply:
Right. Maybe just put a comment on it.

Reply:
Ok, looking more into it, the API change wouldn't be too problematic. The ABI would remain the same, and since the const would determine our usage of the argument, it shouldn't cause an issue with users. However, I misremembered what `ion_string_assign_cstr` did, and it does not copy the string. So, it would have to drop the const anyway inside the function in order to pass it to the `ION_STRING`.

So now my question is: what is the intended use of `ION_STRING`? Looking at the API, most of the functions look like the string is expected to be immutable. If that is the case then `ION_STRING`'s `value` field could be changed to a `const BYTE *value` (pointer to constant BYTE) so that the immutable nature of the string is communicated, and then the const-correctness could bubble up to the other functions.

That will probably require a lot more investigation into how the API is used.

Reply:
Yea. I briefly pondered why `ION_STRING` couldn't be const. Honestly I don't know.